src/index.js

```diff
@@ -1,32 +1,41 @@
-var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest
 
-var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+const API = 'https://rickandmortyapi.com/api/character/'
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
-};
+const fetchData = (url_api, obj = {}) => {
+    return new Promise((resolve, reject) => {
+        const xhttp = new XMLHttpRequest()
+        xhttp.onreadystatechange = event => {
+            if (xhttp.readyState === 4) {
+                if (xhttp.status == 200) resolve({
+                    data: JSON.parse(xhttp.responseText),
+                    store: obj
+                })
+                else return reject({
+                  status: xhttp.status,
+                  url: url_api
+              })
+            }
+        }
 
-fetchData(API, function (error1, data1) {
-  if (error1) return console.error('Error' + ' ' + error1);
-  console.log('Primer Llamado...')
-  fetchData(API + data1.results[0].id, function (error2, data2) {
-    if (error2) return console.error(error1);
-    console.log('Segundo Llamado...')
-    fetchData(data2.origin.url, function (error3, data3) {
-      if (error3) return console.error(error3);
-      console.log('Tercero Llamado...')
-      console.log('Personajes:' + ' ' + data1.info.count);
-      console.log('Primer Personaje:' + ' ' + data2.name);
-      console.log('Dimensión:' + ' ' + data3.dimension);
-    });
-  });
-});
\ No newline at end of file
+        xhttp.open('GET', url_api, false)
+        xhttp.send()
+    })
+}
+
+fetchData(API)
+    .then( ({ data }) => {
+        console.log('Primer Llamado...')
+        return fetchData(`${API}${data.results[0].id}`, { count: data.info.count })
+    } )
+    .then( ({ data, store }) => {
+        console.log('Segundo Llamado...')
+        return fetchData(data.origin.url, { ...store, name:data.name })
+    } )
+    .then( ({ data, store }) => {
```

Comment:
Tener cuidado con los espacios de mas.

```suggestion
    .then(({ data, store }) => {
```